### ISSUE TITLE
Simple Lights Crash

### ISSUE THREAD

**Issue by ghost** [britzl/publicexamples#13]
Crash on Run;

```
ERROR:CRASH: CALL STACK:

# 0 pc   0x3e0a85bd SharedFileList _sigtramp+29
# 1 pc     0x10eba8 dmengine _ZN8dmRender4DrawEPNS_13RenderContextEPNS_9PredicateEPNS_19NamedConstantBufferE+392
# 2 pc     0x10e6e1 dmengine _ZN8dmRender14DrawRenderListEPNS_13RenderContextEPNS_9PredicateEPNS_19NamedConstantBufferE+2545
# 3 pc     0x1135dd dmengine _ZN8dmRender13ParseCommandsEPNS_13RenderContextEPNS_7CommandEj+733
# 4 pc     0x11807a dmengine _ZN8dmRender26UpdateRenderScriptInstanceEPNS_20RenderScriptInstanceEf+170
# 5 pc       0x4aff dmengine _ZN8dmEngine4StepEPNS_6EngineE+975
# 6 pc      0x7d946 dmengine _ZN10dmGraphicsL24OpenGLRunApplicationLoopEPvPFvS0_EPFiS0_E+38
# 7 pc       0x58f1 dmengine _ZN8dmEngine6LaunchEiPPcPFvPNS_6EngineEPvES6_S4_+177
# 8 pc       0x5d1f dmengine _Z11engine_mainiPPc+143
```

**Comment by NaughtyMoleGames** [comment]
> Crash on Run;
> 
> ```
> ERROR:CRASH: CALL STACK:
> 
> # 0 pc   0x3e0a85bd SharedFileList _sigtramp+29
> # 1 pc     0x10eba8 dmengine _ZN8dmRender4DrawEPNS_13RenderContextEPNS_9PredicateEPNS_19NamedConstantBufferE+392
> # 2 pc     0x10e6e1 dmengine _ZN8dmRender14DrawRenderListEPNS_13RenderContextEPNS_9PredicateEPNS_19NamedConstantBufferE+2545
> # 3 pc     0x1135dd dmengine _ZN8dmRender13ParseCommandsEPNS_13RenderContextEPNS_7CommandEj+733
> # 4 pc     0x11807a dmengine _ZN8dmRender26UpdateRenderScriptInstanceEPNS_20RenderScriptInstanceEf+170
> # 5 pc       0x4aff dmengine _ZN8dmEngine4StepEPNS_6EngineE+975
> # 6 pc      0x7d946 dmengine _ZN10dmGraphicsL24OpenGLRunApplicationLoopEPvPFvS0_EPFiS0_E+38
> # 7 pc       0x58f1 dmengine _ZN8dmEngine6LaunchEiPPcPFvPNS_6EngineEPvES6_S4_+177
> # 8 pc       0x5d1f dmengine _Z11engine_mainiPPc+143
> ```

Was there ever a resolution to this?

**Comment by NaughtyMoleGames** [comment]
It has to do with the flicker part of the render script. 

Commenting out the below works.
```
local constants = render.constant_buffer()
constants.time = vmath.vector4(math.random(90, 100) / 100)
```

Then adjust the line below to remove constants
```
render_to_world(self, { self.light_pred })
```

**Comment by britzl** [comment]
I tried the example with the constant buffer intact and I'm not getting a crash. Are you testing with the project exactly like it is on GitHub? Which OS are you using?

**Comment by DrunkenImpGameStudio** [comment]
OS X

- just tested on a windows build and it indeed works fine.

**Comment by NaughtyMoleGames** [comment]
OSX (Mac) - fresh from git with no changes.

I tried it on a windows machine and it indeed does work fine. So it's a Mac only issue.

**Comment by britzl** [comment]
> OSX (Mac) - fresh from git with no changes.
> 
> I tried it on a windows machine and it indeed does work fine. So it's a Mac only issue.

Interesting. I tested on macOS as well.

**Comment by NaughtyMoleGames** [comment]
With the new update to orthocamera - this needs to be updated as well ( ortho helper is no longer used, and multi camera support isn't implemented)

I'd also suggest removing the flicker in render and just use an additional script on the lighted GO, this allows different flickers per light etc ( this is how I have it implemented)

**Comment by britzl** [comment]
> With the new update to orthocamera - this needs to be updated as well

Well, the example uses a specific version of Orthographic so it is not affected by recent changes to the camera.

**Comment by britzl** [comment]
I've updated the example to use the latest version of Orthographic

**Comment by NaughtyMoleGames** [comment]
> I've updated the example to use the latest version of Orthographic

You rock!

**Comment by NaughtyMoleGames** [comment]
Looks like the GUI isn't clearing correctly. It doesn't clear old ui, until you change collections.

[Example](https://imgur.com/4uZmbzt)

**Comment by britzl** [comment]
Fixed in ortho 3.0.2
